### PR TITLE
Derive args from case class in JobTest

### PR DESCRIPTION
**Pull request by regadas**
This is a proposal for #1567

### CODE REVIEW

**Comment by jto** [comment]
This looks unnecessarily complex to me. 

We should be able to just pass the `MinimalWordCountTypedArguments.Arguments("in.txt", "out.txt")` since that's what the pipeline expect to get instead of going  `Arguments -> String -> Arguments`. 

wdyt ?

**Comment by regadas** [comment]
@jto 😄 hm, if I'm looking at this right, I don't think we can get way without the conversion to `String` (at least without too many changes). `scio` jobs don't expect that much setup: 

```scala
object Job {
  def main(args: Array[String])
}
```

And because of this `JobTest` does the following:

https://github.com/spotify/scio/blob/3ca4405c612b84b32be91c16e0961b3808855536/scio-test/src/main/scala/com/spotify/scio/testing/JobTest.scala#L218-L234

Bottom line is, to allow passing a case class without conversion we need to consider rethinking how we setup the job.

File: scio-macros/src/main/scala/com/spotify/scio/testing/ArgGenerator.scala [hunk 1]
```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2016 Spotify AB.
```
Comment (ClaireMcGinty) [inline scio-macros/src/main/scala/com/spotify/scio/testing/ArgGenerator.scala]: outdated copyright year?

**Comment by regadas** [comment]
yup, let's park this for now!

**Pull request closed without merging**